source/features/index.tsx

```diff
@@ -237,7 +241,7 @@ const add = async (id: FeatureID, ...loaders: FeatureLoader[]): Promise<void> =>
 		}
 
 		document.addEventListener('pjax:end', () => {
-			if (repeatOnBackButton || !select.exists('has-rgh')) {
+			if (repeatOnBackButton || !select.exists('has-rgh') || forceReinit) {
```

Comment:
As this line suggests, this is equivalent to `repeatOnBackButton: true`, just under a different name.

This issue is a little complex. Basically we have 2 solutions:

- rework how the automatic deduplication mechanism work (hard)
- rename `repeatOnBackButton: true` to `iAlreadyHandleDeduplicationThankYouVeryMuch: true` and handle deduplication via the usual `:not(.rgh-etc)` selector like we do for most features.

I think the latter is easier. This way we still keep `has-rgh` as a generic fallback but also have the opportunity to opt-out when it causes problems (like in this case)

Reply:
This is exactly what I was asking about. Figured out the root of the issue, but I wasn't exactly sure how we want to tackle it - I think that I'll work on this `dedup` mechanism, and create another PR for it then.